am-lib/src/main/java/uk/gov/hmcts/reform/amlib/ValidationAspect.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.hmcts.reform.amlib;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.hibernate.validator.HibernateValidator;
+import org.hibernate.validator.messageinterpolation.ParameterMessageInterpolator;
+
+import java.lang.reflect.Method;
+import java.util.Set;
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.executable.ExecutableValidator;
+
+import static java.util.stream.Collectors.joining;
+
+@Aspect
+public class ValidationAspect {
+
+    private final ExecutableValidator executableValidator = Validation.byProvider(HibernateValidator.class)
+        .configure()
+        .messageInterpolator(new ParameterMessageInterpolator())
+        .buildValidatorFactory()
+        .getValidator()
+        .forExecutables();
+
+    @Pointcut("execution(public * uk.gov.hmcts.reform.amlib.*Service.*(..))")
+    public void publicServiceMethods() {
+        // No-OP
```

Comment:
did we not agree to throw a not supported operation exception in these cases?

Reply:
Removed that method ;)